system/jlib/jutil.cpp

```diff
@@ -3347,25 +3347,27 @@ const char * generatePassword(StringBuffer &pwd, int pwdLen)
 
     //Ensures each character group used at least once
     srand(time(0));
-    pwd.append(alphaUC[rand() % sizeof(alphaUC)]);
-    pwd.append(alphaLC[rand() % sizeof(alphaLC)]);
-    pwd.append(numeric[rand() % sizeof(numeric)]);
-    pwd.append(symbol[rand() % sizeof(symbol)]);
+    pwd.append(alphaUC[rand() % (sizeof(alphaUC) - 1)]);
```

Comment:
Will this ever output a 'Z'?

Reply:
Yes, before this fix it was outputting NULL, resulting in a truncated password